### PR TITLE
Added content type parsing to TypeScript types. [#858]

### LINKED ISSUE #858: addContentTypeParser and hasContentTypeParser missing from TypeScript definitions
#### What are you trying to achieve or the steps to reproduce?

Being able to write:

```typescript
fastify.addContentTypeParser('multipart', () => {})
```

in a Typescript file

#### What was the result you received?

`Property addContentTypeParser does not exist on type FastifyInstance``

#### What did you expect?

No errors

### PR DESCRIPTION
**Pull request by ShogunPanda**
Added content type parsing to TypeScript types.
Fixes #858.

### CODE REVIEW

**Comment by mcollina** [comment]
Can you please add a unit test? We have some validation for the types.

File: test/types/index.ts [hunk 1]
```diff
@@ -269,6 +269,14 @@ server.setSchemaCompiler(function (schema: object) {
 
 server.addSchema({})
 
+server.addContentTypeParser('foo/bar', {}, (req, done) => {
+  done!(null, {})
+})
+
+server.addContentTypeParser('foo/bar', {}, async (req: http2.Http2ServerRequest) => [])
+
+server.hasContentTypeParser('foo/bar')
```
Comment (allevo) [inline test/types/index.ts]: can you add a check for testing that the returned value is a boolean? (if there's a way to do that)

Comment (ShogunPanda) [reply test/types/index.ts]: Done

Comment (allevo) [reply test/types/index.ts]: @ShogunPanda I'm not able to see that test. Could you point it to me?

Comment (ShogunPanda) [reply test/types/index.ts]: @allevo Sorry, I forgot to push the changes. Now it's there.

**Comment by delvedor** [comment]
Can we merge this?
cc @evanshortiss

**Pull request merged**